src/modules/chat_deleted_messages/index.js

```diff
@@ -0,0 +1,96 @@
+
+const CHAT_CONTAINER = '.chat-room__container';
+const CHAT_LINE_SELECTOR = '.chat-line__message';
+const CHAT_LINE_DELETED_CLASS = 'chat-line__message--deleted';
+
+function chatOberserverHandler(mutations) {
+    const showDeleted = settings.get('showDeletedMessages');
+    const hideDeleted = settings.get('hideDeletedMessages');
+    if (!showDeleted && !hideDeleted) {
+        // return early
+        return;
+    }
+    const elementsToReattach = [];
+    const elementsToRemove = [];
+
+    mutations.forEach(mutation => {
+        if (mutation.type === 'childList' && mutation.target.matches(CHAT_LINE_SELECTOR)) {
+            mutation.addedNodes.forEach(node => {
+                if (!node.__is_prevented && node.classList && node.classList.contains(CHAT_LINE_DELETED_CLASS)) {
+                    elementsToRemove.push([mutation.target, node]);
+                    node.__is_prevented = true; // prevent further mutations to affect the previous list
+                }
+            });
+            mutation.removedNodes.forEach(node => {
+                if (!node.__is_prevented) {
+                    node.__is_prevented = true;
+                    elementsToReattach.push([mutation.target, node]);
+                }
+            });
+        }
+    });
+
+    if (showDeleted) {
+        elementsToRemove.forEach(el => el[1].remove());
+    }
+
+    if (hideDeleted) {
+        elementsToRemove.forEach(el => el[0].remove());
+    }
+
+    if (showDeleted && !hideDeleted) {
+        if (elementsToRemove.length > 0) {
+            elementsToReattach.forEach(([target, node]) => {
+                target.appendChild(node);
+                target.style.opacity = 0.5;
+            });
+        }
+    }
+}
+
+class ChatDeletedMessagesModule {
+    constructor() {
+        settings.add({
+            id: 'showDeletedMessages',
+            name: 'Show Deleted Messages',
+            defaultValue: false,
+            description: 'Turn this on to change <message deleted> back to users\' messages.'
+        });
+        settings.add({
+            id: 'hideDeletedMessages',
+            name: 'Remove Deleted Messages',
+            defaultValue: false,
+            description: 'Completely removes timed out messages from view'
+        });
+        watcher.on('load.chat', () => this.preventMessagesDeleted());
+        this.chatObserver = new MutationObserver(chatOberserverHandler);
+    }
+
+    preventMessagesDeleted() {
+        this.patchClearChat();
+
+        this.chatObserver.disconnect(); // stop listening to old chats mutations
+        this.chatObserver.observe(document.querySelector(CHAT_CONTAINER), {
+            childList: true,
+            subtree: true
+        });
+    }
+
+    patchClearChat() {
+        const chatBuffer = twitch.getChatController().chatBuffer;
+        const consumeEvent = chatBuffer.consumeChatEvent;
+        chatBuffer.consumeChatEvent = function(event) {
+            if (event.type === 17) {
```

Comment:
event types are defined as a constant in the twitch util.. should use that instead

Reply:
Those constants seems to be out of date as 17 is chat clear right now, and it's 16 in the twitch util.